### PR TITLE
test: fix lighting unit test TestBatchSplitRegionByRanges and a panic issue

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
- Make the function call count atomic since the `BatchSplitRegions` method is now called in multiple routines.
- Do not try to flush and unsafe ingest closed engine.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No release note

### CODE REVIEW

**Comment by glorv** [comment]
release-4.0 branch is fix in #957 at [915afe9](https://github.com/pingcap/br/pull/957/commits/915afe9332386c3e8a408f5b6fd615dbb9adb0f7)

File: pkg/lightning/backend/local/local.go [hunk 1]
```diff
@@ -1123,9 +1130,15 @@ func (local *local) CloseEngine(ctx context.Context, engineUUID uuid.UUID) error
 	}
 
 	engineFile := engine.(*File)
-	engineFile.rLock()
+	engineFile.lock(importMutexStateClose)
+	defer engineFile.unlock()
+	if engineFile.closed.Load() {
+		return nil
+	}
+	engineFile.closed.Store(true)
```
Comment (kennytm) [inline pkg/lightning/backend/local/local.go:1138]:
```suggestion
	if engineFile.closed.Swap(true) {
		return nil
	}
```

**Review by kennytm — COMMENTED** [review]
rest LGTM

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**